packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -936,21 +934,23 @@ impl<'m> RunnerImpl<'m> {
         self.flush_batch(
             mv8,
             agent_changes,
-            state_proxy
+            &mut state_proxy
                 .agent_pool_mut()
                 .batch_mut(i_proxy)
-                .ok_or_else(|| format!("Could not access batch at index {i_proxy}"))?,
+                .ok_or_else(|| format!("Could not access batch at index {i_proxy}"))?
```

Comment:
Just to clarify that it's not the group index
```suggestion
                .ok_or_else(|| format!("Could not access proxy at index {i_proxy}"))?
```